TESTDATA_GUIDELINES.md

```diff
@@ -0,0 +1,277 @@
+# Guidelines for creating test data
+<!-- TOC -->
+* [Guidelines for creating test data](#guidelines-for-creating-test-data)
+    * [Terminology](#terminology)
+    * [Multiple apps](#multiple-apps)
+  * [0. Why make your own test data?](#0-why-make-your-own-test-data)
+    * [Ground truth](#ground-truth)
+    * [Completeness](#completeness)
+    * [The GDPR (or similar privacy legislation)](#the-gdpr-or-similar-privacy-legislation)
+    * [Exotic applications](#exotic-applications)
+  * [1. Plan ahead](#1-plan-ahead)
+    * [Investigate the application](#investigate-the-application)
+    * [Create a script](#create-a-script)
+    * [Consider using self-descriptive data](#consider-using-self-descriptive-data)
+    * [Generating large amounts of realistic data](#generating-large-amounts-of-realistic-data)
+  * [2. Creating the test data](#2-creating-the-test-data)
+    * [Creating test data with Puma](#creating-test-data-with-puma)
+    * [Creating test data manually](#creating-test-data-manually)
+  * [3. Extracting and storing the test data](#3-extracting-and-storing-the-test-data)
+    * [Extraction](#extraction)
+    * [Storing the test data](#storing-the-test-data)
+  * [4. Conclusion](#4-conclusion)
+<!-- TOC -->
+
+At the NFI we have been making reference data for some time now, and through the years we discovered some best practices
+improving our process but also the quality of the test data itself. If you are also using Puma to create reference or
+test data, this document will help you to create your datasets better and faster.
+
+### Terminology
+
+In this document, the terms test data and reference data are used interchangeably. While they aren't the exact same, the
+difference is in the use of the data, not the contents of it. Since this document is only about how we create this data,
+in this document the terms can be considered synonymous.
+
+### Multiple apps
+
+In this document we often talk about 'your application', and the assumption seems to be made that only one application
+is involved in the creation of your test data. Of course, this isn't necessarily the case: You might want to create
+reference data involving multiple applications.
+
+If this is the case, this document still applies: anywhere we mention 'your application' consider this as 'your one or
+more applications'. Puma is perfectly able to operate multiple apps at the same time, and all principles considered in
+this document apply to multiple applications just as well as they do to one.
+
+## 0. Why make your own test data?
```

Comment:
While I understand why you have added this section, it seems very extensive and a little defensive (like someone is saying that this library is useless, but people probably don't think that if they've gotten this far). Maybe just make this a list of reasons why we make our own data, and leave it at that. Or maybe put some of this in the README, to explain why Puma exists?